server/app/views/admin/programs/ProgramStatusesView.java

```diff
@@ -334,8 +348,38 @@ private Modal makeStatusUpdateModal(
     Messages messages = messagesApi.preferred(request);
     ProgramStatusesForm formData = form.value().get();
 
+    Boolean isCurrentDefault = formData.getDefaultStatus().orElse(false);
+    LabelTag defaultCheckbox =
+        label()
+            .with(
+                input()
+                    .withType("checkbox")
+                    .withName(ProgramStatusesForm.DEFAULT_CHECKBOX_NAME)
+                    .withCondChecked(isCurrentDefault)
+                    .withClasses(
+                        BaseStyles.CHECKBOX,
+                        "default-checkbox-" + formData.getConfiguredStatusText()),
+                span("Set as default status"));
+    Optional<StatusDefinitions.Status> currentDefaultStatus =
+        program.toProgram().getDefaultStatus();
+    String defaultStatusText =
+        currentDefaultStatus.isPresent() ? currentDefaultStatus.get().statusText() : "";
+    String messagePart =
+        currentDefaultStatus.isPresent() ? "from " + defaultStatusText + " to " : "to ";
+    // I'm so sorry
+    String confirmJs =
+        "var dontShow="
+            + isCurrentDefault.toString()
+            + ";var statusValue=document.querySelector('.input-status-text-"
+            + formData.getConfiguredStatusText()
+            + " .flex input').value;if(!dontShow && document.querySelector('.default-checkbox-"
+            + formData.getConfiguredStatusText()
+            + "').checked){return confirm(\"The default status will be updated "
+            + messagePart
+            + "\"+statusValue+\". Are you sure?\")}else{return true}";
```

Comment:
**suggestion (blocking):** I'd recommend storing whatever data you need from the server as data attributes on the form and then accessing them in JS.

server side:

```java
form()
  .withClasses("status-delete-form")
  .withData("statusText", formData.getConfiguredStatusText())
  .withData("dontShow", isCurrentDefault.toString())
  .withData("messagePart", messagePart)
  // etc
```

client side:

```typescript
import {addEventListenerToElements, assertNotNull} from './util'

addEventListenerToElements(".status-delete-form", "submit", confirmStatusDelete);

function confirmStatusDelete(event: Event) {
  const deleteForm = event.target as HTMLFormElement

  const statusText = assertNotNull(event.dataset.statusText)
  const dontShow = assertNotNull(event.dataset.dontShow) === "true"
  const messagePart = assertNotNull(event.dataset.messagePart)
  const checked = assertNotNull(document.querySelector(`.default-checkbox-${statusText}`) as HTMLInputElement).checked

  if (!dontShow && checked) {
    event.preventDefault();
    if (confirm(confirmationMessage)) {
      deleteForm.submit()
    }
  }
}
```

Reply:
Thanks, this is exactly what I was looking for, much better!